File: pkg/PKG_class.cs
using System;
namespace PKG
{
    public static class PkgGenMd5
    {
        public const string value = "cbccc319ed09be93c69184ff1bf22cc9"; 
    }

    public partial class Player : xx.Object
    {
        public int id;
        public string name;
        public Scene owner;

        public override ushort GetPackageId()
        {
            return xx.TypeId<Player>.value;
        }

        public override void ToBBuffer(xx.BBuffer bb)
        {
            bb.Write(this.id);
            bb.Write(this.name);
            bb.Write(this.owner);
        }

        public override void FromBBuffer(xx.BBuffer bb)
        {
            bb.Read(ref this.id);
            bb.readLengthLimit = 0;
            bb.Read(ref this.name);
            bb.Read(ref this.owner);
        }
        public override void ToString(System.Text.StringBuilder s)
        {
            if (__toStringing)
            {
        	    s.Append("[ \"***** recursived *****\" ]");
        	    return;
            }
            else __toStringing = true;

            s.Append("{ \"pkgTypeName\":\"Player\", \"pkgTypeId\":" + GetPackageId());
            ToStringCore(s);
            s.Append(" }");

            __toStringing = false;
        }
        public override void ToStringCore(System.Text.StringBuilder s)
        {
            s.Append(", \"id\":" + id.ToString());
            if (name != null) s.Append(", \"name\":\"" + name.ToString() + "\"");
            else s.Append(", \"name\":nil");
            s.Append(", \"owner\":" + (owner == null ? "nil" : owner.ToString()));
        }
        public override string ToString()
        {
            var sb = new System.Text.StringBuilder();
            ToString(sb);
            return sb.ToString();
        }
        public override void MySqlAppend(System.Text.StringBuilder sb, bool ignoreReadOnly)
        {
        }
    }
    public partial class Scene : xx.Object
    {
        public xx.List<xx.Ref<Player>> players;

        public override ushort GetPackageId()
        {
            return xx.TypeId<Scene>.value;
        }

        public override void ToBBuffer(xx.BBuffer bb)
        {
            bb.Write(this.players);
        }

        public override void FromBBuffer(xx.BBuffer bb)
        {
            bb.readLengthLimit = 0;
            bb.Read(ref this.players);
        }
        public override void ToString(System.Text.StringBuilder s)
        {
            if (__toStringing)
            {
        	    s.Append("[ \"***** recursived *****\" ]");
        	    return;
            }
            else __toStringing = true;

            s.Append("{ \"pkgTypeName\":\"Scene\", \"pkgTypeId\":" + GetPackageId());
            ToStringCore(s);
            s.Append(" }");

            __toStringing = false;
        }
        public override void ToStringCore(System.Text.StringBuilder s)
        {
            s.Append(", \"players\":" + (players == null ? "nil" : players.ToString()));
        }
        public override string ToString()
        {
            var sb = new System.Text.StringBuilder();
            ToString(sb);
            return sb.ToString();
        }
        public override void MySqlAppend(System.Text.StringBuilder sb, bool ignoreReadOnly)
        {
        }
    }
    public static class AllTypes
    {
        public static void Register()
        {
            xx.Object.RegisterInternals();
            xx.Object.Register<Player>(3);
            xx.Object.Register<Scene>(4);
            xx.Object.Register<xx.List<xx.Ref<Player>>>(5);
        }
    }
}

File: pkg/PKG_class.lua

PKG_PkgGenMd5_Value = 'cbccc319ed09be93c69184ff1bf22cc9'

PKG_Player = {
    typeName = "PKG_Player",
    typeId = 3,
    Create = function()
        local o = {}
        o.__proto = PKG_Player
        o.__index = o
        o.__newindex = o
		o.__isReleased = false
		o.Release = function()
			o.__isReleased = true
		end


        o.id = 0 -- Int32
        o.name = null -- String
        o.owner = null -- PKG_Scene
        return o
    end,
    FromBBuffer = function( bb, o )
        local ReadObject = bb.ReadObject
        o.id = bb:ReadInt32()
        o.name = ReadObject( bb )
        o.owner = ReadObject( bb )
    end,
    ToBBuffer = function( bb, o )
        local WriteObject = bb.WriteObject
        bb:WriteInt32( o.id )
        WriteObject( bb, o.name )
        WriteObject( bb, o.owner )
    end
}
BBuffer.Register( PKG_Player )
PKG_Scene = {
    typeName = "PKG_Scene",
    typeId = 4,
    Create = function()
        local o = {}
        o.__proto = PKG_Scene
        o.__index = o
        o.__newindex = o
		o.__isReleased = false
		o.Release = function()
			o.__isReleased = true
		end


        o.players = null -- List_Ref_PKG_Player_
        return o
    end,
    FromBBuffer = function( bb, o )
        o.players = bb:ReadObject()
    end,
    ToBBuffer = function( bb, o )
        bb:WriteObject( o.players )
    end
}
BBuffer.Register( PKG_Scene )
List_Ref_PKG_Player_ = {
    typeName = "List_Ref_PKG_Player_",
    typeId = 5,
    Create = function()
        local o = {}
        o.__proto = List_Ref_PKG_Player_
        o.__index = o
        o.__newindex = o
		o.__isReleased = false
		o.Release = function()
			o.__isReleased = true
		end

        return o
    end,
    FromBBuffer = function( bb, o )
		local len = bb:ReadUInt32()
        local f = BBuffer.ReadObject
		for i = 1, len do
			o[ i ] = MakeRef( f( bb ) )
		end
    end,
    ToBBuffer = function( bb, o )
        local len = #o
		bb:WriteUInt32( len )
        local f = BBuffer.WriteObject
        for i = 1, len do
			f( bb, o[ i ].Lock() )
		end
    end
}
BBuffer.Register( List_Ref_PKG_Player_ )